rust/lance/src/index/vector/ivf/io.rs

```diff
@@ -28,29 +33,45 @@ pub(super) async fn write_index_partitions(
     writer: &mut dyn Writer,
     ivf: &mut Ivf,
     shuffler: &Shuffler,
+    existing_partitions: Option<&IVFIndex>,
 ) -> Result<()> {
     for part_id in 0..ivf.num_partitions() as u32 {
+        let mut pq_array = Vec::<Arc<dyn Array>>::new();
+        let mut row_id_array = Vec::<Arc<dyn Array>>::new();
+
+        if let Some(existing_idx) = existing_partitions.as_ref() {
+            let part = existing_idx.load_partition(part_id as usize).await?;
+            let pq_idx = part.as_any().downcast_ref::<PQIndex>().unwrap();
+            if pq_idx.code.is_some() {
+                let pq_code_arr = pq_idx.code.as_ref().unwrap().clone();
+                let pq_code_fixed_size_arr = FixedSizeListArray::try_new_from_values(
+                    pq_code_arr.as_ref().clone(),
+                    pq_idx.num_sub_vectors as i32,
+                )?;
+                pq_array.push(Arc::new(pq_code_fixed_size_arr));
+                println!("pq_idx.row_ids: {:#?}", pq_idx.row_ids.as_ref());
```

Comment:
```suggestion
```
Or change to a log

Reply:
Oh this was a left over for debugging. removed it.